numpy/_core/src/multiarray/stringdtype/utf8_utils.h

```diff
@@ -8,8 +8,16 @@ extern "C" {
 NPY_NO_EXPORT size_t
 utf8_char_to_ucs4_code(const unsigned char *c, Py_UCS4 *code);
 
-NPY_NO_EXPORT int
-num_bytes_for_utf8_character(const unsigned char *c);
+static inline int num_bytes_for_utf8_character(const unsigned char *c)
+{
+    // adapted from https://github.com/skeeto/branchless-utf8
```

Comment:
this uses the "unlicense", which according to the FSF is compatible with the GPL, so it's definitely compatible with our BSD license.